compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -668,6 +669,8 @@ private BIRAnnotationValue createAnnotationValue(BLangExpression expr) {
                 return createAnnotationValue(((BLangTypeConversionExpr) expr).expr);
             case STATEMENT_EXPRESSION:
                 return createAnnotationRecordValue((BLangStatementExpression) expr);
+            case INVOCATION:
```

Comment:
Isn't this method only called for constants? What kind of invocations do we get here for? I assume the return type has to be a constant?

Reply:
Here also, I'm not sure if working with `((BLangInvocation) expr).expr` is the correct approach.

Reply:
This happens due to desugar changes invocation is `cloneReadOnly`

Reply:
I am trying to go through to actual expression we wrap by `cloneReadOnly`. I think it is safe to say that `INVOCATION` is not supported.

Reply:
I think here and in `isCompileTimeAnnotationValue` we should specifically check if the invocation name is `cloneReadOnly` and do what you've done only if that is the case.

Reply:
Shall we track this by an issue.. ?

Reply:
Fixing here
https://github.com/ballerina-platform/ballerina-lang/pull/24179